WeatherForecast/show_weather/views.py

```diff
@@ -0,0 +1,54 @@
+import requests
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render, redirect
+from django.views.generic import View
+
+from show_weather import forms
+from show_weather.models import Weather
+
+
+@login_required
+def weather_base_view(request):
+    return render(request, "show_weather/weather_base.html")
+
+
+class SubmitLocationView(View):
+    template_name = "show_weather/form.html"
+
+    def get(self, request):
+        form = forms.LocationForm()
+        return render(request, "show_weather/form.html", {"form": form})
+
+    def post(self, request):
+        form = forms.LocationForm(request.POST)
+        if form.is_valid():
+            form = form.cleaned_data
+            location = form["location"]
+            location = location.lower().capitalize()
+            return redirect("todays_weather", location)
+        return render(request, "show_weather/form.html", {"form": form})
+
+
+class ShowWeatherView(View):
+    def get(self, request, location):
+        response = requests.get(f"https://wttr.in/{location}?format=j1")
```

Comment:
zrób sobie osobny plik constants.py w którym będziesz trzymać stałe, i wtedy możesz do takiej stałej przypisać
```python
WTTR_URL = "https://wttr.in/"
RESPONSE_FORMAT = "j1"
```
a potem w widoku ten url budować tak
```python
response = requests.get(f"{WTTR_URL}{location}?format={RESPONSE_FORMAT}")
```
To jest bardziej eleganckie, nie lubimy zahardcode'owanych rzeczy